### PR TITLE
[stdlib] Add _forEachField(of:options:body:) function

### PR DESCRIPTION
**Pull request by natecook1000**
This function walks all the fields of a struct, class, or tuple type, and calls `body` with the name, offset, and type of each field. `body` can perform any required work or validation, returning `true` to continue walking fields or `false` to stop immediately.

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/ReflectionMirror.swift [hunk 1]
```diff
@@ -160,3 +180,69 @@ extension Mirror {
 #endif
   }
 }
+
+/// Options for calling `_forEachField(of:options:body:)`.
+public // SPI(Foundation)
+struct _EachFieldOptions: OptionSet {
```
Comment (broadwaylamb) [inline stdlib/public/core/ReflectionMirror.swift]: Does it need an availability annotation?

**Comment by natecook1000** [comment]
Still looking at exactly how this function should land. Closing for now…

**Pull request closed without merging**